cli/pkg/config_manager/config_manager.go

```diff
@@ -0,0 +1,286 @@
+package config_manager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/chanzuckerberg/happy/cli/templates"
+	"github.com/chanzuckerberg/happy/shared/config"
+	"github.com/chanzuckerberg/happy/shared/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Service struct {
+	Name           string
+	ServiceType    string
+	Context        string
+	DockerfilePath string
+	Port           int
+	Uri            string
+	Priority       int
+}
+
+const (
+	serviceTypePrivate  = "Service is exposed to the internet, and can only be consumed by other services in the stack (PRIVATE)"
+	serviceTypeExternal = "Service is exposed to the internet (EXTERNAL)"
+	serviceTypeInternal = "Service is exposed to the internet, but is protected by Okta (INTERNAL)"
+)
+
+var (
+	defaultRegions = []string{"us-east-1", "us-east-2", "us-west-1", "us-west-2"}
+
+	serviceTypeMapping = map[string]string{
+		serviceTypePrivate:  "PRIVATE",
+		serviceTypeExternal: "EXTERNAL",
+		serviceTypeInternal: "INTERNAL",
+	}
+
+	ErrServiceSkipped = errors.New("service was skipped")
+)
+
+type happyConfigDescriptor struct {
+	bootstrapConfig *config.Bootstrap
+	happyConfig     *config.HappyConfig
+
+	defaultServicePort string
+	dockerPaths        []string
+	appName            string
+	profiles           []string
+	environmentNames   []string
+	environments       map[string]config.Environment
+	services           []Service
+}
+
+type configAssembler func(ctx context.Context, descriptor *happyConfigDescriptor) (*happyConfigDescriptor, error)
```

Comment:
```suggestion
type configAssembler func(ctx context.Context, descriptor *happyConfigDescriptor) (error)
```
not that important, but seems redundant to pass point to descriptor and return the same descriptor